packages/sdk/src/DeterministicDeployer.ts

```diff
@@ -62,13 +63,21 @@ export class DeterministicDeployer {
     const neededBalance = BigNumber.from(DeterministicDeployer.deploymentGasLimit).mul(DeterministicDeployer.deploymentGasPrice)
     if (bal.lt(neededBalance)) {
       const signer = this.signer ?? this.provider.getSigner()
-      await signer.sendTransaction({
+      const txResponse = await signer.sendTransaction({
         to: DeterministicDeployer.deploymentSignerAddress,
         value: neededBalance,
         gasLimit: DeterministicDeployer.deploymentGasLimit
       })
+      const receipt = await this.provider.waitForTransaction(txResponse.hash, 1, DeterministicDeployer.txWaitTimeout)
```

Comment:
I think the issue is not waiting for "success", but just waiting for result.
For that, adding `then(ret=>ret.wait())` would be enough (in all places)